### PR TITLE
fix: support resumable uploads for nodejs-storage

### PR DESCRIPTION
**Pull request by shaffeeullah**
This PR contains 2 changes required for resumable uploads to work with the node storage client library:
1. Support the request data being an empty object, `{}`.
2. Support the content range being specified in the following format: `bytes 0-*/*`

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6332)

### CODE REVIEW

**Comment by coryan** [comment]
We are working towards making the build output readable, meanwhile:


```diff
diff --git a/google/cloud/storage/emulator/emulator.py b/google/cloud/storage/emulator/emulator.py
index 14355f1..a771b0f 100644
--- a/google/cloud/storage/emulator/emulator.py
+++ b/google/cloud/storage/emulator/emulator.py
@@ -716,7 +716,7 @@ def resumable_upload_chunk(bucket_name):
         x_upload_content_length = int(
             upload.request.headers.get("x-upload-content-length", 0)
         )
-        if (chunk_last_byte == '*'):
+        if chunk_last_byte == "*":
             x_upload_content_length = len(upload.media)
             chunk_last_byte = len(upload.media) - 1
         else:
diff --git a/google/cloud/storage/emulator/gcs/holder.py b/google/cloud/storage/emulator/gcs/holder.py
index 8902a1b..2dc5ac8 100644
--- a/google/cloud/storage/emulator/gcs/holder.py
+++ b/google/cloud/storage/emulator/gcs/holder.py
@@ -61,7 +61,7 @@ class DataHolder(types.SimpleNamespace):
     def init_resumable_rest(cls, request, bucket):
         name = request.args.get("name", "")
         rest_only = {}
-        if len(request.data) > 0 and request.data.decode('utf-8') != "{}":
+        if len(request.data) > 0 and request.data.decode("utf-8") != "{}":
             if name != "":
                 utils.error.invalid("name argument in non-empty payload", None)
             data = json.loads(request.data)
```

File: google/cloud/storage/emulator/utils/common.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 import utils
 
 re_remove_index = re.compile(r"\[\d+\]+|^[0-9]+")
-content_range_split = re.compile(r"bytes (\*|[0-9]+-[0-9]+)\/(\*|[0-9]+)")
+content_range_split = re.compile(r"bytes (\*|[0-9]+-\*|[0-9]+)\/(\*|[0-9]+)")
```
Comment (coryan) [inline google/cloud/storage/emulator/utils/common.py:26]: I think this needs to be:

```py
  content_range_split = re.compile(r"bytes (\*|[0-9]+-[0-9]+|[0-9]+-\*)\/(\*|[0-9]+)")
```

**Review by coryan — CHANGES_REQUESTED** [review]
For some reason the Kokoro builds are not starting, that is weird.  In any case, looks like with these changes the C++ client now crashes the emulator. Could be a bug in the C++ client, but we need to fix it before we can merge these changes.

```
4/61 Test #386: storage_curl_resumable_upload_session_integration_test ..........***Failed    0.07 sec
Running main() from gmock_main.cc
[==========] Running 6 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 6 tests from CurlResumableUploadIntegrationTest
[ RUN      ] CurlResumableUploadIntegrationTest.Simple
../google/cloud/storage/tests/curl_resumable_upload_session_integration_test.cc:63: Failure
Value of: response
Expected: code is equal to OK and message is anything
  Actual: 960-byte object <0E-00 00-00 00-00 00-00 F0-DB 1C-02 00-00 00-00 43-02 00-00 00-00 00-00 43-02 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 ... 4E-33 BA-8C 38-AF FA-9D 9A-2F 78-DF E6-4A 15-E8 6E-BB FA-EF 96-96 B8-2D 76-7A 68-2A ED-8A C3-0A 9E-D1 E1-3C AF-98 CD-AF 15-8A 8B-15 06-98 C1-CB F5-BE B9-97 DF-18 93-81 79-4C 91-F5 90-C8 D8-7A>, whose status is "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.9/site-packages/flask/app.py\", line 1950, in full_dispatch_request\n    rv = self.dispatch_request()\n  File \"/usr/local/lib/python3.9/site-packages/flask/app.py\", line 1936, in dispatch_request\n    return self.view_functions[rule.endpoint](**req.view_args)\n  File \"/workspace/google/cloud/storage/emulator/emulator.py\", line 686, in resumable_upload_chunk\n    items = list(utils.common.content_range_split.match(content_range).groups())\nAttributeError: 'NoneType' object has no attribute 'groups'\n"
 [UNAVAILABLE], with a code that isn't equal to OK, but a message that is anything
2021-04-23T17:22:12.299118929Z [INFO] <139963379994816> Enabling logging for RawClient functions (../google/cloud/storage/client_options.cc:198)
[  FAILED  ] CurlResumableUploadIntegrationTest.Simple (10 ms)
```

**Comment by coryan** [comment]
Please rebase to the latest version at `master`? You are missing some scripts needed to run the builds, and I am hoping that will wake up the Kokoro builds too.

**Comment by shaffeeullah** [comment]
@coryan Thanks so much for your help getting the checks to pass!! Really appreciate it :)

**Review by coryan — APPROVED** [review]

**Comment by coryan** [comment]
I changed the commit message, we typically filter out things about the emulator in our release notes by labeling all its messages `test: `

**Comment by coryan** [comment]
Should we delete the branch?

**Pull request merged**